.github/workflows/release.yml

```diff
@@ -0,0 +1,34 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "*"
+
+jobs:
+  release:
+    name: release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout commit and fetch tag history
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
So this ended up being a heck of a hassle. We're using `use_scm_version=True` in `setup.py`, which will [attempt to infer the correct version number from Git history](https://github.com/pypa/setuptools_scm/#default-versioning-scheme). This is different from `pystac`, which [checks](https://github.com/azavea/pystac/blob/develop/setup.py#L6) a manually incremented [`versions.py`](https://github.com/azavea/pystac/blob/develop/pystac/version.py) file.

If this workflow is only run when a new tag is pushed, then that version should always resolve to the value of the tag, which is what we want.

Check out the version in [this build output](https://github.com/azavea/django-ecsmanage/pull/22/checks?check_run_id=883502666) though. It thinks that it's creating the first release, `0.1`, because it can't find any tag history.

Turns out that the `checkout` action retrieves a shallow clone by default. By setting [`fetch-depth: 0`](https://github.com/actions/checkout#fetch-all-history-for-all-tags-and-branches) we fetch the full history of the repo, resulting in [this build output](https://github.com/azavea/django-ecsmanage/pull/22/checks?check_run_id=883504715). It's resolving to the next dev version (`1.1.1`) with Git metadata attached, consistent with the behavior described in the `use_scm_version` documentation.

I'm confident this change will cause `use_scm_version` to resolve to the right tag if it were a tagged commit. I tested this locally by running `git tag` and then `python setup.py --version` to see what `use_scm_version` would resolve for the version number. Also, I have the full revision history available in my local repo, so it's able to resolve to the correct version.

Reply:
Interesting. I think this tradeoff makes sense and the fix is consistent with what other folks are doing to solve the same problem. The Python publishing examples I saw that use `use_scm_version` and don't reference `fetch-depth` all appear to be using `checkout@v1`, which was apparently less efficient in its checkout process.

Reply:
Yup, looks like v2 introduced a shallow fetch by default: https://github.com/actions/checkout/blob/main/CHANGELOG.md#v2-beta